src/server/wesnothd/server.hpp

```diff
@@ -110,7 +111,7 @@ class server : public server_base
 	{
 		std::string ip;
 		int attempts;
-		std::time_t first_attempt;
+		std::chrono::system_clock::time_point first_attempt;
```

Comment:
Looks like making it steady_clock would involve changing `ban_manager::ban` to accept steady time points as well.

Reply:
It's used to check if there have been too many failed login attempts. This should not be influenced if the clock is adjusted.

Where is `first_attempt` used as a ban time? I guess you mean the `now` used with it is needed as system clock for the ban? Unless there is a way to convert I suppose you need both, yes.

Reply:
Indeed, the `now` that is used to check if the ban has expired is then used to set the ban time. The logic is basically, if now + default_ban_time < now, then remove ban, and if the number of attempts is too high, then ban *until* now + default_ban_time. That last timestamp is a system time_point.

I guess there's a bit of a problem with the way the ban manager holds times. Right now, it's designed to home a system time_point representing the ban end. That, however, means bans are subject to the whims of the clock instead of being for a fixed steady duration. It's not really possible to mix both.

For the login attempt ban, I could just note both a steady and a system timestamp and use the latter to calculate the ban end time, but it would be better to sort out whether we want bans to be of a strict duration or not.

Reply:
In the interest of getting this merged, I'm going to leave it as system_clock for now since that matches the existing `std::time` behavior and handle further steady_clock refactoring later.